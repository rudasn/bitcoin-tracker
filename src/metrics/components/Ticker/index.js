import React from 'react'

import ChangeColoredText from '../ChangeColoredText'
import TrendArrow from '../TrendArrow'

export default ({ change=0, value=null, percentage=0 }) =>
  <div className="ticker">
    <TrendArrow change={ change }/>
    <span>{ value } </span>
    <ChangeColoredText change={ change }>
      { change.toFixed(2) } ({ percentage.toFixed(2) }%)
    </ChangeColoredText>
  </div>
