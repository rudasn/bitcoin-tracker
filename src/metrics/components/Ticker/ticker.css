.ticker {
  color: var(--color-dark);
}
