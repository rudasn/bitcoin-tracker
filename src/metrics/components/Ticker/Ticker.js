import React from 'react'
import propTypes from 'prop-types'

import './ticker.css'
import ChangeColoredText from '../ChangeColoredText'
import TrendArrow from '../TrendArrow'

const Ticker = ({ change=0, value=null, percentage=0 }) =>
  <div className="ticker">
    <TrendArrow change={ change }/>
    <span>{ value } </span>
    <ChangeColoredText change={ change }>
      { change.toFixed(2) } ({ percentage.toFixed(2) }%)
    </ChangeColoredText>
  </div>

Ticker.propTypes = {
  change: propTypes.number,
  percentage: propTypes.number,
  value: propTypes.node,
}

export default Ticker
