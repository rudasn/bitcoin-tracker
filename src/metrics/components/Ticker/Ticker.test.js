import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Ticker from './Ticker'

it('renders without crashing', () => {
  const element = <Ticker />
  const div = document.createElement('div')
  ReactDOM.render(element, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders no change correctly', () => {
  const element = <Ticker change={ 0 } percentage={ 0 } value={ 0 } />
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders positive correctly', () => {
  const element = <Ticker change={ 1 } percentage={ 1 } value={ 1 } />
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders negative correctly', () => {
  const element = <Ticker change={ -1 } percentage={ -1 } value={ -1 } />
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})
