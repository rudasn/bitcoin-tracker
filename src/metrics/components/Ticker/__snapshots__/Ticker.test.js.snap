// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders negative correctly 1`] = `
<div
  className="ticker"
>
  <span
    className="change-colored-text-down"
  >
    <span
      className="arrow arrow-down"
    />
  </span>
  <span>
    -1
     
  </span>
  <span
    className="change-colored-text-down"
  >
    -1.00
     (
    -1.00
    %)
  </span>
</div>
`;

exports[`renders no change correctly 1`] = `
<div
  className="ticker"
>
  <span
    className="change-colored-text"
  >
    <span
      className="arrow"
    />
  </span>
  <span>
    0
     
  </span>
  <span
    className="change-colored-text"
  >
    0.00
     (
    0.00
    %)
  </span>
</div>
`;

exports[`renders positive correctly 1`] = `
<div
  className="ticker"
>
  <span
    className="change-colored-text-up"
  >
    <span
      className="arrow arrow-up"
    />
  </span>
  <span>
    1
     
  </span>
  <span
    className="change-colored-text-up"
  >
    1.00
     (
    1.00
    %)
  </span>
</div>
`;
